src/Common/src/CoreLib/System/BitConverter.cs

```diff
@@ -19,10 +19,23 @@ public static class BitConverter
         // This field indicates the "endianess" of the architecture.
         // The value is set to true if the architecture is
         // little endian; false if it is big endian.
+#if MONO
+		[Intrinsic]
+		public static readonly bool IsLittleEndian;
+
+		static BitConverter () {
+			unsafe {
+				ushort i = 0x1234;
+				byte *b = (byte*)&i;
+				IsLittleEndian = (*b == 0x34);
+			}
+		}
+#else
```

Comment:
It'd be better not to touch CoreFX code and just if-def this out completely for #if MONO and make the type partial with Mono implementation in mcs/class/System/BitConverter.cs